src/cleo/io/inputs/token_parser.py

```diff
@@ -37,8 +37,7 @@ def _parse(self) -> list[str]:
 
                 continue
 
-            if self._current is not None:
```

Comment:
are you sure about this change? I am hesitant to believe that...

Reply:
The entire snippet is:
```py
        while self._current is not None:
            if self._current.isspace():
                # Skip spaces
                self._next()

                continue

            if self._current is not None:
                tokens.append(self._parse_token())
```
The `while` and the `if` have identical conditions and the `while` body wouldn't be entered if the condition wasn't satisfied + there's no way its truthiness changes when executing the body in a synchronous scenario.

Reply:
If you take a look at what `self._next()` is doing, you might see that it actually changes what `self._current` is, so I believe that it changes the logic.

Reply:
Yes but there's a `continue` right after it.